### PR TITLE
feat(CI): add more github actions

### PR DESCRIPTION
**Pull request by xwartz**
- Add GitHub Action to assign labels to PRs based on title
- Add GitHub Action to automerge dependency updates from Dependabot

### CODE REVIEW

File: .github/workflows/generate-labels.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    types: [opened, edited, synchronize]
+
+jobs:
+  label:
+    name: 'Label PR based on title'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: srvaroa/labeler@v0.8
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
```
Comment (Shadowfiend) [inline .github/workflows/generate-labels.yml:14]: Can you help me understand what this brings to the table? Adding labels based on info already searchable from the title doesn't seem immediately useful on its face, but very possible I'm missing something.

File: .github/workflows/merge-dependencies.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    if: github.actor == 'dependabot[bot]'
+    steps:
+      - name: 'Checkout repository'
+        uses: actions/checkout@v2.3.4
+      - name: 'Automerge dependency updates from Dependabot'
+        uses: ahmadnassri/action-dependabot-auto-merge@v2.4.0
+        # Guarantee that commit comes from Dependabot (don't blindly trust external GitHub Actions)
+        if: github.actor == 'dependabot[bot]'
+        with:
+          github-token: ${{ github.token }}
```
Comment (Shadowfiend) [inline .github/workflows/merge-dependencies.yml:23]: I'm not sure I'm comfortable auto merging dependency updates, at least not yet—much as it pains me. The potential for supply chain attacks (not to mention introducing bugs) with something automated like dependabot isn't something I feel we've fully explored yet.

**Review by Shadowfiend — COMMENTED** [review]
Thank you for the contribution! Had a few questions and maybe a thing or two we need to discuss on the team before sending these through due to the nature of the changes; left my questions below 😊

File: .github/workflows/generate-labels.yml (hunk 1 above)
Comment (xwartz) [reply .github/workflows/generate-labels.yml:14]: We can use labels to manage and filter PRs, not only to help better search, but also to help reviewers distinguish PR categories at a glance, just like issues.

File: .github/workflows/merge-dependencies.yml (hunk 2 above)
Comment (xwartz) [reply .github/workflows/merge-dependencies.yml:23]: ok, I understand what you mean, let's remove the automatic merge.

**Comment by mhluongo** [comment]
@xwartz I'm closing this for now — we don't want to auto-merge for security reasons, and we prefer users and devs input issue labels themselves rather than doing it via a title and an action that tries to guess their intent.

I imagine we'll want to reconsider that last decision down the road though.

Thanks for the suggestion and keep them coming!

**Pull request closed without merging**